climpyrical/cmd/preprocess_model.py

```diff
@@ -0,0 +1,169 @@
+"""
+quick usage of climpyrical preprocess_mode.py
+usage:
+python preprocess_model.py -i input.nc -o output.nc -m True
+
+This script takes a CanRCM4 model at the native resolution and
+downscales from 50 km to  5 km and fills in missing land values
+using external masks.
+"""
+
+from climpyrical.data import read_data, gen_dataset, interpolate_dataset
+from climpyrical.gridding import regrid_ensemble, extend_north
+
+import click
+from pkg_resources import resource_filename
+import logging
+
+import warnings
+
+import numpy as np
+
+warnings.filterwarnings("ignore")
+
+
+@click.command()
+@click.option("-i", "--in-path", help="Input CanRCM4 file", required=True)
+@click.option("-o", "--out-path", help="Output file", required=True)
+@click.option(
+    "-m", "--fill-glaciers", help="Refill glacier points", default=True
+)
+@click.option(
+    "-l",
+    "--log-level",
+    help="Logging level",
+    type=click.Choice(["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]),
+    default="INFO",
+)
+def run_processing(in_path, out_path, fill_glaciers, log_level):
+    """Completes the preprocessing of the model required
+    for the NRC project.
+    Args:
+        in_path, out_path (strings): directories of NetCDF4 file
+            input and output. Must give filename, too with extension
+            .nc. Overwites files with same name in same directory.
+        fill_glaciers (bool): whether to fill spurious glacier
+            points with preprocessed mask. Default is True.
+        quiet (bool): whether to log quietly or not
+    Returns:
+        Creates a NetCDF4 file at out_path at target resolution
+    """
+    logging.basicConfig(level=log_level)
+
+    ds = read_data(in_path)
+    dv = list(ds.data_vars)[0]
+
+    rlon, rlat = np.meshgrid(ds.rlon, ds.rlat)
+    mean = ds[dv].values
+    kelvin = 273.15  # K
+
+    logging.info("Detect unit conversions")
```

Comment:
What if it's *already* Kelvin? Or on Fahrenheit? I'd recommend actually checking the `units` attribute before doing a blind unit conversion. In most of our other software we actually use a units library to check the source and destination units and convert, but that's probably overkill for this purpose.

Reply:
Hadn't considered that, I didn't realize that we keep track of this because it's only visible in the raw data field itself. :+1:

Reply:
Just a note on this:

We currently don't have data standards on incoming CanRCM4 models (something that will be discussed soon with others, and hopefully resolved from here on out).

I.e. there are many mislabelled unit attributes. I'm hesitant (for deadline reasons) to try and account for several new and unlikely input units (Fahrenheit), but will fix the CanRCM4 models that are mislabelled and leave room for a simple check in `read_data`  if it comes up down the road, and make it clear to the users that a units attirbute of `degC` _will_ automatically be converted to `K`.